File: src/main/java/chapter4/etc/TimeIntervalExample.java
package chapter4.etc;

import common.CommonUtils;
import common.Log;
import io.reactivex.Observable;
import io.reactivex.schedulers.Timed;


public class TimeIntervalExample {

    public void marbleDiagram() {
        String[] data = {"1","3","7"};

        CommonUtils.exampleStart();
        Observable<Timed<String>> source = Observable.fromArray(data)
                .delay(item -> {
                    CommonUtils.doSomething();
                    return Observable.just(item);
                })
                .timeInterval();

        source.subscribe(Log::it);
        CommonUtils.sleep(1000);
    }

    public static void main(String[] args) {
        TimeIntervalExample demo = new TimeIntervalExample();
        demo.marbleDiagram();
    }

}

File: src/main/kotlin/chapter4/etc/TimeIntervalExample_.kt
package chapter4.etc

import common.CommonUtils_
import common.Log_
import io.reactivex.Observable


class TimeIntervalExample_ {

    fun marbleDiagram() {
        val data: Array<String> = arrayOf("1", "3", "7")

        CommonUtils_.exampleStart()
        val source = Observable.fromArray(*data)
                .delay {
                    CommonUtils_.doSomething()
                    Observable.just(it)
                }
                .timeInterval()

        source.subscribe { Log_.it(it) }
        CommonUtils_.sleep(1000)
    }

}


fun main(args: Array<String>) {
    val demo = TimeIntervalExample_()
    demo.marbleDiagram()
}
